### PR TITLE
test: add tests for AddTrip

### LINKED ISSUE #47: Test Add Trip Screen
This issue aims to test the functionalities of the AddTripScreen.

### PR DESCRIPTION
**Pull request by mirceagavrizi**
## Summary

Added 4 tests for the AddTrip and a setter for the trip type in TripsViewModel 

## Checklist

Ensure all / most of these are checked before the pull request is submitted.
- [x] This PR resolves #47 
- [x] Tests have been added for new functionality.
- [ ] Screenshots or UI changes have been attached (if applicable).
- [ ] Documentation has been updated (if applicable).

##  Testing

- **Manual Testing:**
    - [ ] Tested on emulator / physical device.
- **Unit Tests**
    - [x] Added units tests for this

If you haven't added tests, please create a new issue and assign it.

##  Related Issues/Tickets

Closes #47

### CODE REVIEW

**Comment by apana03** [comment]
Please make sure to reach 80 % code coverage

File: app/src/androidTest/java/com/android/voyageur/ui/overview/AddTripTest.kt [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    composeTestRule.onNodeWithTag("inputStartDate").performTextInput("notadate")
+
+    composeTestRule.onNodeWithTag("inputEndDate").performTextClearance()
+    composeTestRule.onNodeWithTag("inputEndDate").performTextInput("05/01/2024")
+
+    composeTestRule.onNodeWithTag("tripSave").performClick()
+
+    verify(tripRepository, never()).updateTrip(anyNonNull(), anyNonNull(), anyNonNull())
+  }
+
```
Comment (1andreea26) [inline app/src/androidTest/java/com/android/voyageur/ui/overview/AddTripTest.kt:82]: Here you could also add a test to check that the user can't submit with an empty name for the trip.

File: app/src/main/java/com/android/voyageur/ui/overview/AddTrip.kt [hunk 2]
```diff
@@ -192,7 +198,7 @@ fun AddTripScreen(
                     }
                   },
                   enabled = name.isNotBlank() && startDate.isNotBlank() && endDate.isNotBlank(),
```
Comment (1andreea26) [inline app/src/main/java/com/android/voyageur/ui/overview/AddTrip.kt:200]: I think you should also add a Toast requesting the user to enter a name for the trip in case they leave it empty (like in the bootcamp).

File: app/src/androidTest/java/com/android/voyageur/ui/overview/AddTripTest.kt [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  fun addTripScreen_tripTypeSelection() {
+    composeTestRule.setContent { AddTripScreen(tripsViewModel, navigationActions) }
+
+    composeTestRule.onNodeWithTag("tripTypeBusiness").performClick()
+    assert(tripsViewModel.tripType.value == TripType.BUSINESS)
+
+    composeTestRule.onNodeWithTag("tripTypeTourism").performClick()
+    assert(tripsViewModel.tripType.value == TripType.TOURISM)
+  }
+}
```
Comment (1andreea26) [inline app/src/androidTest/java/com/android/voyageur/ui/overview/AddTripTest.kt:93]: I checked out the code coverage report and noticed the timestamp is directly converted in the AddTripScreen. I would suggest making a separate method to convert the timestamp, and then writing a test for that method. Or at least writing a test which checks that the timestamp is correctly converted.

**Review by 1andreea26 — CHANGES_REQUESTED** [review]
The tests look great, but please check out the suggestions to improve code coverage :)

**Review by apana03 — APPROVED** [review]
No major comments, only a code refactoring one. Congrats!

**Review by apana03 — APPROVED** [review]

**Review by 1andreea26 — APPROVED** [review]
The new tests are great. I will approve :).
